### PR TITLE
Disable local users for a smooth single-idp mode

### PR DESCRIPTION
**Pull request by braginini**
## Describe your changes

Add LocalAuthDisabled option to embedded IdP configuration                                                                                                                    
                                                                                                                                                                                
  This adds the ability to disable local (email/password) authentication when using the embedded Dex identity provider. When disabled, users can only authenticate via external 
  identity providers (Google, OIDC, etc.).                  

This simplifies user login when there is only one external IdP configured. The login page will redirect directly to the IdP login page.                                                                                                                    
                                                                                                                                                                                
  Key changes:                                                                                                                                                                  
  - Added LocalAuthDisabled field to EmbeddedIdPConfig                                                                                                                          
  - Added methods to check and toggle local auth: IsLocalAuthEnabled, HasNonLocalConnectors, DisableLocalAuth, EnableLocalAuth                                                  
  - Validation prevents disabling local auth if no external connectors are configured                                                                                           
  - Existing local users are preserved when disabled and can login again when re-enabled                                                                                        
  - Operations are idempotent (disabling already disabled is a no-op)

## Issue ticket number and link

## Stack

<!-- branch-stack -->

### Checklist
- [ ] Is it a bug fix
- [ ] Is a typo/documentation fix
- [x] Is a feature enhancement
- [ ] It is a refactor
- [ ] Created tests that fail without the change (if possible)

> By submitting this pull request, you confirm that you have read and agree to the terms of the [Contributor License Agreement](https://github.com/netbirdio/netbird/blob/main/CONTRIBUTOR_LICENSE_AGREEMENT.md).

## Documentation
Select exactly one:

- [x] I added/updated documentation for this change: https://github.com/netbirdio/docs/pull/583
- [ ] Documentation is **not needed** for this change (explain why)

### Docs PR URL (required if "docs added" is checked)
Paste the PR link from https://github.com/netbirdio/docs here:

https://github.com/netbirdio/docs/pull/__


<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

* **New Features**
  * Runtime/config option to disable local (email/password) authentication; surfaces local_auth_disabled in account settings/API and UI.

* **Behavior / Safety**
  * Prevent disabling local auth unless another identity connector exists; startup and account flows block local user creation/invite when disabled.
  * Explicit enable/disable workflows for local auth and settings propagation to runtime configuration.

* **Tests**
  * New tests covering startup, disabled-state behavior, API responses, and edge cases.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

### CODE REVIEW

File: management/server/instance/manager.go [hunk 1]
```diff
@@ -104,13 +104,22 @@ func NewManager(ctx context.Context, store store.Store, idpManager idp.Manager)
...
 	users, err := m.embeddedIdpManager.GetAllAccounts(ctx)
 	if err != nil {
 		return err
 	}
+	hasLocalUsers := len(users) > 0
 
 	m.setupMu.Lock()
-	m.setupRequired = len(users) == 0
+	m.setupRequired = !(hasAccounts || hasLocalUsers)
 	m.setupMu.Unlock()
```
[the remark this answers is not in the document]
Comment (braginini) [reply management/server/instance/manager.go:123]: This is not possible as you can have LocalAuthDisabled=true if there is no account

File: management/server/idp/embedded.go [hunk 2]
```diff
@@ -192,11 +199,32 @@ func NewEmbeddedIdPManager(ctx context.Context, config *EmbeddedIdPConfig, appMe
 		return nil, err
 	}
 
+	log.WithContext(ctx).Debugf("initializing embedded Dex IDP with config: %+v", config)
```
[the remark this answers is not in the document]
Comment (braginini) [reply management/server/idp/embedded.go:202]: irrelevant
